test/unit/components/navpatientheader/index.test.js

```diff
@@ -0,0 +1,180 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { BrowserRouter } from 'react-router-dom';
+import _ from 'lodash';
+
+import NavPatientHeader from '../../../../app/components/navpatientheader';
+
+const expect = chai.expect;
+
+const defaultPatientProps = { 
+  userid: 1234,
+  profile: { 
+    fullName: 'Vasyl Lomachenko',
+    patient: { birthday: '2010-10-20', mrn: '567890' }
+  }
+};
+
+describe.only('NavPatientHeader', () => {
+  const mockHistory = { push: sinon.stub() }
+  const mockTrackMetric = sinon.stub();
+
+  NavPatientHeader.__Rewire__('useHistory', sinon.stub().returns(mockHistory));
+
+  afterEach(() => {
+    mockTrackMetric.reset();
+    mockHistory.push.reset();
+  });
+
+  describe(('visibility of patient info and actions'), () => {
+    context('personal user with non-root permissions', () => {
+      const props = { // personal user with non-root permissions;
+        user: { roles: [] },
+        trackMetric: mockTrackMetric,
+        patient: { ...defaultPatientProps, permissions: { root: false } },
+      };
```

Comment:
For some reason I struggled immensely with having just a default prop object at the top and then making a copy of it per test. I tried `_.cloneDeep` and all sorts of things.